src/geometry.rs

```diff
@@ -486,6 +509,29 @@ impl From<(u32, u32, u32)> for TriangleFace {
     }
 }
 
+impl PartialEq for TriangleFace {
```

Comment:
I disagree with this (as I already mentioned on our slack discussion), at least until you convince me 😹 . Let's discuss offline.

Reply:
I'm strongly **for** this solution and you'll have a hard time convincing me otherwise ;) . Later tonight I'll write down my reasons.

Reply:
1. The face is indifferent to which vertex/normal pair is referenced first. Faces `(0, 1, 2)`, `(1, 2, 0)`, `(2, 0, 1)` are identical. Not similar, not alike, they really are identical.
2. Our data structure suffers from non-existence of a circular list in the language. Instead, we're compensating it by the `Eq` and `Hash` functions.
3. Another solution for this would be some unification of face-vertex shift. Let's say when the face is being created, the winding will be reordered so that the lowest index will come first in list. I don't like it mainly because there is no need for this.
4. Having a custom `Eq` and `Hash` function allows for direct use of language features, such as `==` or hash-based structures, which is and will be incredibly useful.
5. There is no reason for wrapping the "linear" face into a "circular" face tuplestruct because there is no use case where the linear face would be useful or even correct. This is a different case from the `OrientedEdge` and `UnorientedEdge`, where the different view makes some real difference and there are different use cases for them. The face and its wrapper would only add more complexity and will have no reason, nor use.
6. There are currently no tests that would fail with the custom `Eq` function but if there were, I prefer manually checking for field-by-field equality in the test rather than in the main software logic. It would privilege technicalities to semantics.
7. Because of points 1. and 2. I believe any logic that would consider faces `(0, 1, 2)` and `(1, 2, 0)` different would be wrong.

Reply:
Thanks for the write-up. I hear your arguments and sympathize with them - this is actually very similar to `HashSet` and `HashMap`, which also override/weaken the meaning of `Eq`. The other side is that I don't agree with the following:

> I believe any logic that would consider faces (0, 1, 2) and (1, 2, 0) different would be wrong even in tests.

I think that at least in tests, we do want to have the ability to compare chunks of memory without additional semantics. I will review the existing tests to see how this would affect them.

> There are currently no tests that would fail with the custom Eq

I am worried about the opposite case - a test falsely passing because of the weaker `Eq`.